CSCMETAX-516: [REF] Separate the fetching of oai_dc_urnresolver recor…

junsk1 opened the pull request:
…ds in metax_oai_server to its own code vs. other metadataprefixes. Return data with cursor and batch size applied only after urnresolver records have been checked whether they should be included. Fix oai_dc and oai_datacite metadataprefixes.

hannu40k commented:
You mentioned yesterday that the tests for oai are a bit lacking. Would this refactoring be a good opportunity to remedy the situation a little bit? All the tests are also shoved into a single suite, so some grouping could be in order too.

junsk1 commented:
OAI-PMH tests are quite ok, but what is actually missing tests for e.g. catalog_record to datacite xml conversion, which was one of the reasons why some parts of oai-pmh server had broken. But it is out of the scope of this ticket. I added some missing tests, reorganized the test file and removed unneeded stuff.
